### ISSUE TITLE
Invalid URL when trying to setup PostgreSQL Monitor

### ISSUE THREAD

**Issue by pagubg** [louislam/uptime-kuma#5405]
### 📑 I have found these related issues/pull requests

-

### 🛡️ Security Policy

- [X] I agree to have read this project [Security Policy](https://github.com/louislam/uptime-kuma/security/policy)

### Description

When creating a PostgreSQL monitor, and if the password contains ? character in the connection string URL, the monitor won't work because the password cannot be parsed correctly. 

### 👟 Reproduction steps

Create postgresql somewhere with password which contains ? character: example pw: my?password
Set up in PostgreSQL monitor with this password in the connection URL string and save.
There will be an error message like this: 
`2024-12-04T14:47:37+02:00 [MONITOR] WARN: Monitor #29 'mypostgreSQLmonitor': Failing: Invalid URL | Interval: 60 seconds | Type: postgres | Down Count: 0 | Resend Interval: 0`
Also, the password is not masked when editing the same monitor. 
If there isn't ? character in the password, everything works as expected.

### 👀 Expected behavior

Should pass the check

### 😓 Actual Behavior

Throws error with Invalid URL

### 🐻 Uptime-Kuma Version

1.23.15

### 💻 Operating System and Arch

docker image

### 🌐 Browser

chrome under linux

### 🖥️ Deployment Environment

- Runtime: k8s cluster
- Database: 
- Filesystem used to store the database on: nfs
- number of monitors: < 30

**Comment by CommanderStorm** [comment]
did you URL-encode the password?

**Comment by pagubg** [comment]
Yes. It works with a password without any special characters. The issue could be easily replicated. Just use the "?" character in the password for the PostgreSQL user. Also, some other special chars will replicate it. Same for MySQL, etc.
